rakelib/artifacts.rake

```diff
@@ -62,13 +63,26 @@ namespace "artifact" do
   end
 
   def exclude?(path)
-    excludes.any? { |ex| path == ex || (File.directory?(ex) && path =~ /^#{ex}\//) }
+    excludes.any? { |ex| path_matches_exclude?(path, ex) }
   end
 
-  def files
+  def exclude_oss?(path)
+    path_matches_exclude?(path, "x-pack" ) || exclude?(path)
+  end
+
+  def oss_excluder
+    @oss_excluder ||= self.method(:exclude_oss?)
```

Comment:
while it works, passing around memoised unbounded methods isn't the most straight-forward to understand; let's make a note to take some time on cleaning this up when we're not under pressure timing-wise.

Reply:
Maybe I've been doing FP for too long but it doesn't seem weird to me? What would the cleanup be?

Reply:
I mean, we could make it an option `:exclude_oss` that the various methods are provided with, but I think they read the same. Passing a predicate seems just as readable IMHO